app/src/org/commcare/dalvik/activities/CommCareHomeActivity.java

```diff
@@ -991,163 +943,15 @@ protected void onResume() {
      * Decides if we should actually be on the home screen, or else should redirect elsewhere
      */
     private void attemptDispatchHomeScreen() {
-
-        // Before anything else, see if we're in a failing db state
-        int dbState = CommCareApplication._().getDatabaseState();
-        if (dbState == CommCareApplication.STATE_MIGRATION_FAILED) {
-            CommCareApplication._().triggerHandledAppExit(this,
-                    getString(R.string.migration_definite_failure),
-                    getString(R.string.migration_failure_title), false);
-            return;
-        } else if (dbState == CommCareApplication.STATE_MIGRATION_QUESTIONABLE) {
-            CommCareApplication._().triggerHandledAppExit(this,
-                        getString(R.string.migration_possible_failure),
-                        getString(R.string.migration_failure_title), false);
-            return;
-        } else if (dbState == CommCareApplication.STATE_CORRUPTED) {
-            handleDamagedApp();
-        }
```

Comment:
This db state check logic has been moved to the dispatch activity. That means it is no longer called every time the home screen is resumed, I assume that is a good thing. It will be called every time the dispatch activity is resumed: before app install, before login, before home screen launch.